src/components/Main/SortingSection.js

```diff
@@ -12,13 +13,22 @@ import ErrorMessage from 'components/Common/ErrorMessage';
 import Option from 'components/Common/Option';
 import SortingButton from 'components/Common/SortingButton';
 import styles from 'components/Main/SortingSection.module.css';
+import AddFolderModal from 'components/Modal/AddFolderModal';
+import DeleteFolderModal from 'components/Modal/DeleteFolderModal';
+import EditFolderNameModal from 'components/Modal/EditFolderNameModal';
+import ShareModal from 'components/Modal/ShareModal';
 
 import { FOLDERS_API_URL, LINKS_API_URL } from 'services/api';
 
 function SortingSection({ selectedFolder, setSelectedFolder }) {
   const LOADING_MESSAGE = 'Loading...';
   const ALL = { id: 0, name: '전체' };
 
+  const [hoveredFolder, setHoveredFolder] = useState(null);
+  const [isAddFolderModalOpen, setIsAddFolderModalOpen] = useState(false);
+  const [isOptionListModalOpen, setIsOptionListModalOpen] = useState(false);
```

Comment:
각 컴포넌트마다 모달 상태를 관리하게 되면 지금처럼 모달의 종류마다 상태를 가지게 되는 문제가 발생할 수 있으니 전역으로 모달을 처리하는 방법을 고민해보도록 합시다!

그렇지 않고 만약 컴포넌트마다 관리를 하게 된다면 지금처럼 `isAddFolderModalOpen`, `isOptionsListModalOpen`의 상태를 따로 두지 않고 아래와 같이 모든 modal 상태를 관리하는 상태를 만드는게 관심사를 응집하는 방법 중 하나입니다!
```suggestion
  const [modalOpenState, setModalOpenState] = useState({
    isAddFolderOpen: false,
    isOptionsListOpne: false,
  })
  
  ...
  const handleAddFolderButtonClick = () => {
    setModalOpenState(prev => ({...prev, isAddFolderOpen: true})
  };
  
```